### DISCUSSION TITLE
run Yolo from webcam

### DISCUSSION THREAD

**Discussion by jardaoravec** [ultralytics/yolov5#2425] [General]
Hello I see in read me I should use $ python detect.py --source 0  # webcam for running on webcam. But my question is. 
I want to run this from my IDE- Pycharm, how can I do that ? I want run Yolo from IDE on webcam is it possible? I want to process frames so that is the reason I don't want to run from console.

**Comment by glenn-jocher** [comment]
@jardaoravec these two topics are completely unrelated. 

You can run any python file directly from PyCharm, and PyCharm also has it's own built-in Terminal you can use to run bash commands.

**Reply by jardaoravec** [reply]
@glenn-jocher To run Yolo detection I am running detect.py file. I put video in data/images/ folder and it saves me nicely output video to runs folder. My question is what should I do when I want to run detect.py on webcam . I really appreciate your hard work on this project. Thank you

**Reply by glenn-jocher** [reply]
For webcam inference use source 0:

https://github.com/ultralytics/yolov5/blob/886f1c03d839575afecb059accf74296fad395b6/detect.py#L150
